duckduckgo/Android
# Add Pixels when app restarts in foreground by automatic data clearer

Task/Issue URL: https://app.asana.com/0/0/1180090817363681/f
Tech Design URL: 
CC: 

**Description**:
This PR adds logic to send pixels when the app gets restarted due to automatic data clearer being triggered with the app in foreground.

We will send pixels to detect if the user opened the app from an external intent or without it. This will give us some insights about the impact of this issue:  https://github.com/duckduckgo/Android/issues/780

**Steps to test this PR**:
In order to test this, we need to:
1. Disable WorkManager: comment line AutomaticDataClearer#121 `scheduleBackgroundTimerToTriggerClear(clearWhenOption.durationMilliseconds())`
1. Enable automatic data clearer in settings (5 seconds for testing purposes)

Test 1:
1. Start the app as normal after being inactive for 5 seconds
1. Ensure app restarts due to automatic dataclearer
1. Ensure pixel `mf_r` sent

Test 2:
1. Start the app from a home screen icon or any external link after being inactive for 5 seconds
1. Ensure app restarts due to automatic dataclearer
1. Ensure pixel `mf_ri` sent

Test 3:
1. Start the app from search widget after being inactive for 5 seconds
1. Ensure app restarts due to automatic dataclearer
1. Ensure pixel `mf_ri` sent

Test 3:
1. Start the app from an external link in less than 5 seconds
1. Ensure app didn't restart
1. Send the app to backgound (home button or turn off screen)
1. Bring back the app to foreground after being inactive for 5 seconds
1. Ensure app restarts due to automatic dataclearer
1. Ensure pixel `mf_r` sent



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
